diff --git a/search_api_ranges.module b/search_api_ranges.module
index 8f6004c..b2a2b5b 100644
--- a/search_api_ranges.module
+++ b/search_api_ranges.module
@@ -357,8 +357,14 @@ function search_api_ranges_block_slider_view_form_submit($form, &$form_state) {
   $params = drupal_get_query_parameters($_GET, array('q', 'page'));
 
   // Get pretty path path and goto()
-  list($language, $path) = language_url_split_prefix(request_path(), language_list());
-  $language = $language ? $language : NULL;
+  if (drupal_multilingual() && variable_get('locale_language_negotiation_url_part') == LOCALE_LANGUAGE_NEGOTIATION_URL_PREFIX) {
+    list($language, $path) = language_url_split_prefix(request_path(), language_list());
+    $language = $language ? $language : NULL;
+  }
+  else {
+    $path = request_path();
+    $language = NULL;
+  }
   if (module_exists('facetapi_pretty_paths')) {
     $exists = FALSE;
     $split_path = explode('/', $path);
